hatch/commands/new.py

```diff
@@ -93,18 +95,30 @@ def new(name, no_env, pyname, pypath, global_packages, env_name, basic, cli, lic
     if basic:
         settings['basic'] = True
 
-    if licenses:
-        settings['licenses'] = licenses.split(',')
-
     settings['cli'] = cli
 
     origin = os.getcwd()
-    d = os.path.join(origin, name)
+    package_name = name if name else click.prompt('project name')
+
+    d = os.path.join(origin, package_name)
 
     if os.path.exists(d):
         echo_failure('Directory `{}` already exists.'.format(d))
         sys.exit(1)
 
+    if licenses:
+        settings['licenses'] = map(str.strip, licenses.split(','))
```

Comment:
In `init` this comes after the interactive branch. What do you want to give precedence?

Reply:
If the user runs this: 

```
hatch init -i -l 'mit,mpl'
```

Hatch will use the `mit,mpl` as the default value during the interactive mode. If the user chooses to override it during the interactive mode then the overridden value from the interactive mode will be used.

Reply:
Right now the thing is that in `new` the `if licenses` comes before the `if interactive` but in `init` it's the opposite.

Reply:
Good catch. They were functionally equivalent. But I've changed the code to match the same pattern in both files.